Add docs on how to disable cache provider

nicoddemus opened the pull request:
After discussion in #1029

I suggest we start to advise people who are having problems related to `cache` to use this trick so it doesn't affect their use of `pytest-2.8` and force them to pin down to `pytest-2.7.X`.

RonnyPfannschmidt commented:
can it actually be reenabled?

nicoddemus commented:
Do you mean without removing the option from `pytest.ini` manually? I don't know the order of precedence in this cases, but I would guess command line wins configuration.

hpk42 commented on doc/en/cache.rst:
```diff
@@ -10,6 +10,16 @@ cache: working with cross-testrun state
+      [pytest]
+      addopts = -p no:cacheprovider
+
+
```
i wonder if it's the right thing to advise to do this on a per-project level.  You might want cache behaviour on dev machines but not on CI systems.  Therefore we should rather advise for setting the environment variable ``PYTEST_ADDOPTS`` for respective environments.

hpk42 replied:
in addition we should see to add some info about environment variables with ``py.test --help`` i guess. (like "tox -h" does)

nicoddemus commented:
Added environment variables to `--help` as requested, and moved plugin disabling to `plugins.txt`. I think the more detailed explanation fits there better.

nicoddemus commented:
Can we merge this?

RonnyPfannschmidt commented:
of course, sorry for the delay

nicoddemus commented:
No worries! :smile:
